packages/coin-sol/src/config/types.ts

```diff
@@ -0,0 +1,30 @@
+import { Transport } from '@coolwallet/core';
+// export type Transport;
+
+export type header = {
+  numRequiredSignatures: number;
+  numReadonlySignedAccounts: number;
+  numReadonlyUnsignedAccounts: number;
+};
+
+export type CompliedInstruction = {
+  accounts: number[];
+  data: string;
+  programIdIndex: number;
+};
+
+export type messageType = {
+  header: header;
+  accountKeys: Buffer[];
+  recentBlockhash: string;
+  instructions: CompliedInstruction[];
+};
```

Comment:
The properties of this type might be too abstract.
```suggestion
export type messageType = {
  fromPubKey: string;
  toPubKey: string;
  lamports: number;
};
```

Reply:
We want to make it clearer, but recentBlockHash is the field that needs to be generated by @solana/web3.js, the rest fields like instructions are generated by complex formula, if put into sdk only include from, to , lamports will need to install @solana/web3js to sol/sdk. Should we do that, we'd love to hear from you?

Reply:
`recentBlockHash` should be part of `messageType`, Sorry for missing it.
But the reset fields are generated by https://github.com/solana-labs/solana-web3.js/blob/8b986f250c3897d99858605c2406670a17d92cc2/src/transaction.ts#L206-L374, it should not be complex. 
As for `TransactionInstruction`, it can be generated easily from https://github.com/solana-labs/solana-web3.js/blob/8b986f250c3897d99858605c2406670a17d92cc2/src/system-program.ts#L676-L707 using `@solana/buffer-layout`.

Reply:
We will be using other solution, using compleMessage and move to project, it's several risks. We will push asap which you can check that solution. Thanks for feedback